File: Proyecto/AppWeb/Backend/src/main/java/org/uteq/backend/service/TipoDocumentoService.java
package org.uteq.backend.service;

import org.springframework.stereotype.Service;
import org.uteq.backend.dto.TipoDocumentoDTO;
import org.uteq.backend.repository.TipoDocumentoRepositoryCustom;

import java.util.List;
import java.util.Map;

@Service
public class TipoDocumentoService {

    private final TipoDocumentoRepositoryCustom repo;

    public TipoDocumentoService(TipoDocumentoRepositoryCustom repo) {
        this.repo = repo;
    }

    public List<TipoDocumentoDTO> listar() {
        return repo.listar();
    }

    public Map<String, Object> crear(String nombre, String descripcion, Boolean obligatorio) {
        return repo.crear(nombre, descripcion, obligatorio);
    }

    public Map<String, Object> editar(Long id, String nombre, String descripcion, Boolean obligatorio) {
        return repo.editar(id, nombre, descripcion, obligatorio);
    }

    public Map<String, Object> toggle(Long id) {
        return repo.toggle(id);
    }
}

File: Proyecto/AppWeb/Frontend/src/app/services/TipoDocumento.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface TipoDocumento {
  idTipoDocumento: number;
  nombre:          string;
  descripcion:     string;
  obligatorio:     boolean;
  activo:          boolean;
}

@Injectable({ providedIn: 'root' })
export class TipoDocumentoService {

  private readonly API = `${environment.apiUrl}/tipos-documento`;

  constructor(private http: HttpClient) {}

  listar(): Observable<TipoDocumento[]> {
    return this.http.get<TipoDocumento[]>(this.API);
  }

  crear(data: { nombre: string; descripcion: string; obligatorio: boolean }): Observable<any> {
    return this.http.post(this.API, data);
  }

  editar(id: number, data: { nombre: string; descripcion: string; obligatorio: boolean }): Observable<any> {
    return this.http.put(`${this.API}/${id}`, data);
  }

  toggle(id: number): Observable<any> {
    return this.http.patch(`${this.API}/${id}/toggle`, {});
  }
}
